blog/2018/12/20/mojolicious-serving-angular-web-app/index.markdown

```diff
@@ -0,0 +1,379 @@
+
+    Sachin@02:06 PM[~/workspace/project/mojo_angular/NgDemo]$ ng build --base-href=./
+                                                                                          
+    Date: 2018-12-15T06:06:48.550Z
+    Hash: f3749aba56348b1e51e3
+    Time: 27091ms
+    chunk {main} main.js, main.js.map (main) 10.3 kB [initial] [rendered]
+    chunk {polyfills} polyfills.js, polyfills.js.map (polyfills) 223 kB [initial] [rendered]
+    chunk {runtime} runtime.js, runtime.js.map (runtime) 6.08 kB [entry] [rendered]
+    chunk {styles} styles.js, styles.js.map (styles) 16.3 kB [initial] [rendered]
+    chunk {vendor} vendor.js, vendor.js.map (vendor) 3.35 MB [initial] [rendered]
+    Sachin@02:06 PM[~/workspace/project/mojo_angular/NgDemo]$
+
+ Note that **dist** directory is created with `NgDemo` folder which contains compiled angular app files:
+
+    Sachin@02:10 PM[~/workspace/project/mojo_angular/NgDemo]$ ls
+    README.md         angular.json      dist              e2e               node_modules      package-lock.json package.json      src               tsconfig.json     tslint.json
+    Sachin@02:10 PM[~/workspace/project/mojo_angular/NgDemo]$   
+
+##### b. Copy everything within the folder dist/ to a folder on the `public` directory in Mojolicious app
+Mojo full app consists of `public` directory which is a static file directory (served automatically).
+Copy Angular app compiled into `dist/` to `public` directory of mojo app so that mojo will serve automatically. 
+
+    Sachin@02:12 PM[~/workspace/project/mojo_angular/NgDemo]$ cd dist
+    Sachin@02:13 PM[~/workspace/project/mojo_angular/NgDemo/dist]$ ls NgDemo
+    Sachin@02:13 PM[~/workspace/project/mojo_angular/NgDemo/dist]$cp -R NgDemo ~/workspace/project/mojo_angular/mojo_angular_app/public/
+    Sachin@02:13 PM[~/workspace/project/mojo_angular/NgDemo/dist]
+
+##### c. Modify Mojo Application class to serve angular app:
+Mojo Application class, `~/workspace/project/mojo_angular/mojo_angular_app/lib/MojoAngularApps.pm` in our demo, should be modified
+to push compiled angular app directory to mojo's static paths list so that it is served automatically as well.
+
+    package MojoAngularApp;
+
+    use Mojo::Base 'Mojolicious';
+
+    # This method will run once at server start       
+    sub startup {
+        my $self = shift;
+
+        # Load configuration from hash returned by "my_app.conf"                                                                         
+        my $config = $self->plugin('Config');
+        # Documentation browser under "/perldoc"                                       
+        $self->plugin('PODRenderer') if $config->{perldoc};
```

Comment:
Do recent versions of Mojolicious include this plugin in the default generated app? I don't think so. Can you check and remove if I'm correct? It is just extra noise here.